src/pdf_beamtime/src/pdf_beamtime_server.cpp

```diff
@@ -266,6 +208,46 @@ void PdfBeamtimeServer::remove_obstacles_service_cb(
   planning_scene_interface_.removeCollisionObjects(removable_object);
 }
 
+template<typename RequestT, typename ResponseT>
+void PdfBeamtimeServer::new_obstacle_service_cb(
+  const typename RequestT::SharedPtr request,
+  typename ResponseT::SharedPtr response)
+{
+  try {
+    // Adds the new obstacle name to the existing list
+    auto obj_param = node_->get_parameters({"object_names"})[0].as_string_array();
+    obj_param.push_back(request->name);
+    node_->set_parameters(
+      {rclcpp::Parameter("object_names", obj_param)});
+
+    // Add the common parameters
+    node_->declare_parameter("objects." + request->name + ".type", request->type);
+    const std::vector<std::pair<std::string, double>> parameters = {
+      {"x", request->x}, {"y", request->y}, {"z", request->z}, {"h", request->h}};
+
+    for (const auto & param : parameters) {
+      node_->declare_parameter("objects." + request->name + "." + param.first, param.second);
+    }
+
+    // Checks the message type to differentiate Box type from Cylinder
+    if constexpr (std::is_same_v<RequestT, BoxObstacleMsg::Request>) {
+      node_->declare_parameter("objects." + request->name + ".w", request->w);
+      node_->declare_parameter("objects." + request->name + ".d", request->d);
+    } else if constexpr (std::is_same_v<RequestT, CylinderObstacleMsg::Request>) {
+      node_->declare_parameter("objects." + request->name + ".r", request->r);
+    } else {
+      RCLCPP_INFO(node_->get_logger(), "Message type not registered");
```

Comment:
I'd make this a warning.

Reply:
RCLCPP_INFO is changed to RCLCPP_WARN
https://github.com/maffettone/erobs/blob/82ea0b0722451076fc8c67563b396d2862af8a5a/src/pdf_beamtime/src/pdf_beamtime_server.cpp#L238